### PR TITLE
feat: add lambda to pipe to invalidate cache

### PR DESCRIPTION
**Pull request by TheOrangePuff**
**Description of the proposed changes**  

* Adds an optional lambda function that invalidates the `/graphql` endpoint on a successful mesh deployment 

**Notes to reviewers**  

🛈  When you've finished leaving feedback, please add a final comment to the PR tagging the author, letting them know that you have finished leaving feedback

### CODE REVIEW

File: packages/graphql-mesh-server/assets/handlers/invalidate-cloudfront-cache.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      Paths: {
+        Quantity: paths.length,
+        Items: paths,
+      },
+      CallerReference: "lambda",
+    },
+  };
+
+  const command = new CreateInvalidationCommand(invalidationInput);
+  return await cfClient.send(command);
```
Comment (AdamJHall) [inline packages/graphql-mesh-server/assets/handlers/invalidate-cloudfront-cache.ts:28]: If you're returning the result you don't need to await the result

File: packages/graphql-mesh-server/lib/pipeline.ts [hunk 2]
```diff
@@ -110,6 +115,49 @@ export class CodePipelineService extends Construct {
...
+            __dirname,
+            "../assets/handlers/invalidate-cloudfront-cache.ts"
+          ),
+          description: "Lambda function to invalidate CloudFront cache.",
+          runtime: Runtime.NODEJS_18_X,
+          handler: "index.handler",
+          timeout: Duration.seconds(5),
+          environment: {
+            PATHS: "/graphql",
+            DISTRIBUTION_ID: props.cloudFrontDistributionId || "",
```
Comment (AdamJHall) [inline packages/graphql-mesh-server/lib/pipeline.ts:133]: You don't need the `||` here as you are checking for `cloudFrontDistributionId` above

**Review by AdamJHall — DISMISSED** [review]
Couple of small changes that could be made but don't really affect the result so happy to approve this now :)

File: packages/graphql-mesh-server/assets/handlers/invalidate-cloudfront-cache.ts (hunk 1 above)
Comment (TheOrangePuff) [reply packages/graphql-mesh-server/assets/handlers/invalidate-cloudfront-cache.ts:28]: Oh really? I was worried if I didn't await the lambda might not finish sending the command

File: packages/graphql-mesh-server/lib/pipeline.ts (hunk 2 above)
Comment (TheOrangePuff) [reply packages/graphql-mesh-server/lib/pipeline.ts:133]: Ah yeah, must have added this before I included the if statement

File: packages/graphql-mesh-server/assets/handlers/invalidate-cloudfront-cache.ts (hunk 1 above)
Comment (AdamJHall) [reply packages/graphql-mesh-server/assets/handlers/invalidate-cloudfront-cache.ts]: Nah the function itself is async, so there is an implied await on the result (Could be wrong here but that's my understanding).

**Review by AdamJHall — APPROVED** [review]

**Pull request merged**